deployment/terraform/utils.go

```diff
@@ -240,6 +240,17 @@ func fillConfigTemplate(configTmpl string, data map[string]any) (string, error)
 	return buf.String(), nil
 }
 
+// convertToTerraformVar converts a list parameter to a json encoded string
+func convertToTerraformVar[T any](param T) string {
+	result, err := json.Marshal(param)
+	if err != nil {
+		mlog.Error("failed to convert parameter to terraform var", mlog.Any("param", param), mlog.Err(err))
+		return ""
+	}
+
+	return string(result)
+}
```

Comment:
As per the code, this converts any parameter since it just marshals into json, but happy to make it just a `String()` function on the type.

Reply:
> but happy to make it just a String() function on the type

That's what I did for `TerraformMap`, the type of `CustomTags`. That way you can simply pass it to `Sprintf`. I don't have a super strong opinion on this, but I find it slightly better, since it isolates its function.

Reply:
https://github.com/mattermost/mattermost-load-test-ng/pull/833/commits/4f0a3626163c5fba4245679b2bd9bce2d97f091a